src/engine/source/api/src/catalog/handlers.cpp

```diff
@@ -10,12 +10,42 @@
 namespace api::catalog::handlers
 {
 
+namespace
+{
+
+template<typename r_type>
+auto checkResourcePermission(const base::Name& name,
+                             const std::string& role,
+                             const rbac::IRBAC::AuthFn& authFn,
+                             const rbac::IRBAC::AuthFn& authSystemFn)
+    -> std::optional<decltype(::api::adapter::genericError<r_type>(std::string {}))>
+{
+    auto isSystemResource = name.parts().size() > 0 && name.parts()[0] == "system";
```

Comment:
You could replace that string literal `"system"` with a `constexpr` similar to` constexpr auto ENGINE_NAMESPACE = "user";` of `defaultSettings.hpp`

Reply:
This will be removed in the issue in progress,  Engine - Update internal policy #18525

Reply:
This is RBAC, and will be modified in the implementation of RBAC, for now we do not have the requirements.